mantidproject/mantid
# Fix wrap_pyui.py when building for Python3

The call to `open()` in `buildconfig/wrap_pyui.py` script failed with Python3 as some of the opened files would in fact be in UTF-8 but `open()` would treat them as 7bit ASCII or whatever. This PR fixes building with Python3.

**To test:**

You need to have a GNU/Linux system with Python3 installed.
1. Check that the builds pass. This should take care of Python2 part and serve as a reference.
2. `rm -fr *` your build directory. 
3. Your build directory is now gone.
4. Run cmake with `-DPYTHON_EXECUTABLE=/usr/bin/python3`.
5. Build all. There shouldn't be any problems with Python.

A code review would also be a plus.

Fixes #20220. 

**Release Notes** 

This was an internal change, affecting only the rare case of experimental Python3 builds. *Does not need to be in release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
